### PR TITLE
[#16278] fix tips in white border

### LINKED ISSUE #16278: [Android] Horizontal frames are not centered within the camera box and a bulge appears at the end of the frame line
follow up for https://github.com/status-im/status-mobile/pull/16206#issuecomment-1590861340
Related to comment https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS/Feedback-for-Mobile?type=design&node-id=1929-154137#429224561

### Steps to reproduce:
1. Open 'Sign in by syncing' page
2. Enable the camera

### Actual result:
- The horizontal frames are not in the middle of the camera box
- Bulge is shown at the end of the frame lines
image

### Expected result:
- Framing lines should be in the middle of the camera box
- The end points has to be round, not flat
image

### Device:
- Real device: Pixel 7a, Android 13
- Android Emulator: Pixel 6 API 31. v12

### PR DESCRIPTION
**Pull request by ulisesmac**
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Partial solution to #16278

### Summary

This PR fixes the tips in the white border
image

### Review notes
Sometimes in React Native is hard to perfectly center an UI element due to the unit managed in different devices, the tip in these borders had that problem. I solved it by reducing the width from 1 to 0.9 and it worked in the devices I tested (emulators and physical devices).

#### Platforms

- Android
- iOS

### Steps to test

- Open Status in a fresh install and click on Sign in, or add a new existing Status profile
- Activate the camera
- The border tips are not outside the view

status: ready

### CODE REVIEW

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs [hunk 1]
```diff
@@ -180,55 +176,42 @@
                        view-finder (assoc layout :height (:width layout))]
                    (reset! qr-view-finder view-finder)))}])
 
-(defn- border
-  [border1 border2 corner]
-  [rn/view {:style (style/border border1 border2 corner)}])
+(defn- white-border
+  [corner]
+  (let [border-styles (style/white-border corner)]
```
Comment (ibrkhalil) [inline src/status_im2/contexts/syncing/scan_sync_code/view.cljs:170]: White border is a variable, Why use it as a function?

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs [hunk 2]
```diff
@@ -180,55 +176,42 @@
...
+    [rn/view
+     [rn/view {:style (border-styles :border)}]
```
Comment (ibrkhalil) [inline src/status_im2/contexts/syncing/scan_sync_code/view.cljs:172]: I don't think this is right ..

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs [hunk 3]
```diff
@@ -180,55 +176,42 @@
...
-        [rn/view
-         [border :border-bottom-width :border-right-width :border-bottom-right-radius]
-         [border-tip {:right 0 :top -1}]
-         [border-tip {:left -1 :bottom 0}]]]]
-      [quo/text
-       {:size   :paragraph-2
-        :weight :regular
-        :style  style/viewfinder-text}
-       (i18n/label :t/ensure-qr-code-is-in-focus-to-scan)]]]))
+  (let [layout-size (+ (:width qr-view-finder) 2)]
```
Comment (ibrkhalil) [inline src/status_im2/contexts/syncing/scan_sync_code/view.cljs:188]: I'd destructure width on line 198

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/status_im2/contexts/syncing/scan_sync_code/view.cljs:172]: The function call to border-styles I mean.

Comment (ulisesmac) [reply src/status_im2/contexts/syncing/scan_sync_code/view.cljs:172]: @ibrkhalil 
I am answering this and the previous comment.

Hash maps in Clojure are also functions, which receive a key and will return the value associated.

`({:a 1} :a)` => `1`

Like keywords being functions that extract itself from a map.
`(:a {:a 1})` => `1`

Or, like sets returning the element if it exists:
`(#{:a} :a)` => `:a`

(Actually, sets are hashmaps with the same key and value)

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs (hunk 3 above)
Comment (ulisesmac) [reply src/status_im2/contexts/syncing/scan_sync_code/view.cljs:188]: I think it's not worth it in this case.
Just a key extraction, maybe if there are more keys, it'd be better to destructure. I think both are fine

File: src/status_im2/contexts/syncing/scan_sync_code/view.cljs (hunk 2 above)
Comment (ibrkhalil) [reply src/status_im2/contexts/syncing/scan_sync_code/view.cljs:172]: Yeah, I keep forgetting
The function call to hashmaps will always remain confusing to me :)
Cool

**Review by ibrkhalil — APPROVED** [review]
lgtm

File: src/status_im2/contexts/syncing/scan_sync_code/style.cljs [hunk 4]
```diff
@@ -77,29 +77,37 @@
...
+   :width            78
+   :height           78
+   border-vertical   2
+   border-horizontal 2
+   corner-radius     16})
+
+(def white-border
+  (let [base-tip {:background-color colors/white
+                  :position         :absolute
+                  :height           1.9
```
Comment (J-Son89) [inline src/status_im2/contexts/syncing/scan_sync_code/style.cljs]: it might be worth adding a comment for use of 1.9 here.

**Review by J-Son89 — APPROVED** [review]

**Review by smohamedjavid — APPROVED** [review]

File: src/status_im2/contexts/syncing/scan_sync_code/style.cljs (hunk 4 above)
Comment (ulisesmac) [reply src/status_im2/contexts/syncing/scan_sync_code/style.cljs]: Done :+1:

**Comment by qoqobolo** [comment]
Hey @ulisesmac , thanks for your work!

The tip looks good now, but the frame placement is still different from the design.

Design:

Screenshot 2023-07-24 at 16 00 47

PR:

Screenshot 2023-07-24 at 16 02 20


In the description, you indicate that this PR fixes https://github.com/status-im/status-mobile/issues/16278, but in fact only partially. Could you clarify please, the second part of the problem will be fixed separately, right?
In this case, this PR can be merged, but please make sure that the issue is not closed.

**Comment by ulisesmac** [comment]
@qoqobolo 
The problem is device's DPI, when we add an unit to paddings or margins, they are not always moved equally in all devices, so, these borders are shown correctly in some devices.
We have this problem in many components but it's not always noticeable.

I will not close this issue, and I will reference this comment.

**Comment by qoqobolo** [comment]
@ulisesmac gotcha!
Thanks for the explanation and the fix. Let's merge it!

**Pull request merged**